### PR TITLE
bt-results supports --max-count <number> and --format short

### PR DESCRIPTION
**Pull request by akiellor**

### CODE REVIEW

File: bin/bt-results [hunk 1]
```diff
@@ -26,21 +46,16 @@ Grit.debug = true if opts[:debug]
...
-    $stdout.print(pipeline.to_hash.to_json)
+  pipelines = r.commits(:start => opts[:commit], :max_count => opts[:max_count]).map do |commit|
+    stage_definition = YAML.load(`#{find_command :stages} --commit #{commit.sha} "#{r.path}"`)
+    BT::Pipeline.new(commit, stage_definition)
+  end
+  
+  if format = FORMATS[opts[:format]]
+    $stdout.print(format.call pipelines)
+  else
+    puts "'#{opts[:format]}' is not an acceptable format. Acceptable formats are [#{FORMATS.keys.join("|")}]"
```
Comment (quad) [inline bin/bt-results]: I appreciate it says this AFTER doing the processing. :-D

Comment (akiellor) [inline bin/bt-results]: Hmmm noted.

File: lib/bt/pipeline.rb [hunk 2]
```diff
@@ -24,6 +24,13 @@ module BT
       {commit.sha => stages.inject({}) {|result, stage| result.merge(stage.to_hash)}}
     end
 
+    def status
+      return 'FAIL' if stages.any?(&:fail?)
+      return 'PASS' if stages.all?(&:ok?)
+      return 'IN PROGRESS' if stages.any?(&:ok?)
```
Comment (quad) [inline lib/bt/pipeline.rb]: I've been mulling over "`IN PROGRESS`" for a few days, trying to think of an alternative. I liked that `FAIL` and `PASS` were equal length, one word, and easily parseable (by human and computer).

But I can't think of a good alternative.
- `WAIT`
- `WORK`
- `PLAY`
- `INCOMPLETE`

Blargh!

Comment (akiellor) [inline lib/bt/pipeline.rb]: TBH YAGNI. I think we cut back to FAIL, PASS, INCOMPLETE. Incomplete exists in the code at the moment, defined as not done, so I think we go with that.

**Pull request merged**